src/bootstrap/config/tests.rs

```diff
@@ -178,3 +178,18 @@ fn profile_user_dist() {
     }
     Config::parse_inner(&["check".to_owned()], get_toml);
 }
+
+#[test]
+fn rust_optimize() {
+    assert_eq!(parse("").rust_optimize.is_release(), true);
+    assert_eq!(parse("rust.optimize = false").rust_optimize.is_release(), false);
+    assert_eq!(parse("rust.optimize = true").rust_optimize.is_release(), true);
+    assert_eq!(parse("rust.optimize = \"1\"").rust_optimize.get_opt_level(), Some("1".to_string()));
```

Comment:
i'd prefer to require this to be an int instead of a strong, so `rust.optimize = 1` instead of `\"1\"`. no one is using the new config yet so i think we can break hard here.

Reply:
What about the "s" and "z"? Do you mean we need to use an IntOrString here?

Reply:
yes, exactly

Reply:
I'll fix it later.